libs/api/src/lib/graphql/blocks.ts

```diff
@@ -487,6 +496,269 @@ export const GraphQLPublicTeaser = new GraphQLUnionType({
   ]
 })
 
+export const GraphQLTeaserType = new GraphQLEnumType({
+  name: 'TeaserType',
+  values: {
+    [TeaserType.Article]: {value: TeaserType.Article},
+    [TeaserType.PeerArticle]: {value: TeaserType.PeerArticle},
+    [TeaserType.Event]: {value: TeaserType.Event},
+    [TeaserType.Page]: {value: TeaserType.Page},
+    [TeaserType.Custom]: {value: TeaserType.Custom}
+  }
+})
+
+export const GraphQLTeaserListBlockFilter = new GraphQLObjectType({
+  name: 'TeaserListBlockFilter',
+  fields: {
+    tags: {type: new GraphQLList(new GraphQLNonNull(GraphQLID))}
+  }
+})
+
+export const GraphQLTeaserListBlockFilterInput = new GraphQLInputObjectType({
+  name: 'TeaserListBlockFilterInput',
+  fields: {
+    tags: {type: new GraphQLList(new GraphQLNonNull(GraphQLID))}
+  }
+})
+
+export const GraphQLTeaserListBlock = new GraphQLObjectType<TeaserListBlock, Context>({
+  name: 'TeaserListBlock',
+  fields: {
+    blockStyle: {
+      type: GraphQLString,
+      resolve: resolveBlockStyleIdToName
+    },
+    teaserType: {
+      type: GraphQLTeaserType
+    },
+    filter: {type: new GraphQLNonNull(GraphQLTeaserListBlockFilter)},
+    take: {type: GraphQLInt},
+    skip: {type: GraphQLInt},
+    teasers: {
+      type: new GraphQLNonNull(new GraphQLList(GraphQLTeaser)),
```

Comment:
Reusing the same interface for teasers for convenience and code consistency. Had to keep the optional on teasers else I would need to rename the property in the client schema due to naming conflicts.

I tried to make all teasers optional, but the editor is a bit fucked up how blocks are rendered and reuses the output types for the input and that causes issues. It's quite a refactor and abandoned it halfway through.